compiler/inferring/restriction-isset.cpp

```diff
@@ -158,5 +157,12 @@ bool RestrictionIsset::find_dangerous_isset_dfs(int isset_flags, tinf::Node *nod
 
 bool RestrictionIsset::is_restriction_broken() {
   vector<tinf::Node *> bt;
-  return find_dangerous_isset_dfs(a_->isset_flags, a_, &bt);
+  int isset_flags = a_->isset_flags;
+  kphp_assert(isset_flags);
+  // before dfs, Node::isset_flags contains what ifi mask we need to search for
+  // during dfs, we use the same field to store what mask has already been checked for
+  a_->isset_flags = 0;
```

Comment:
This change seams to be very strange. You fixed flags in one vertex, but a lot of others, which this one can possibly depend, will contain some trash inside. Why this could work?

Reply:
Lazha :( You're right. BTW, it shows, that we have no tests for such cases. 
I've reverted this commit for now. Probably, I'll think how to deal with isset_flags/isset_was and this restriction later on.